pkg/ipamd/ipamd.go

```diff
@@ -613,29 +626,25 @@ func (c *IPAMContext) tryUnassignIPsOrPrefixesFromAll() {
 			for _, toDelete := range ips {
 				// Don't force the delete, since a freeable IP might have been assigned to a pod
 				// before we get around to deleting it.
-				var ipv4Cidr net.IPNet
+
+				ipv4Str := toDelete.IP.String()
 				if c.enableIpv4PrefixDelegation {
-					_, ipv4CidrPtr, err := net.ParseCIDR(toDelete)
-					if err != nil {
-						return
-					}
-					ipv4Cidr = *ipv4CidrPtr
-				} else {
-					ipv4Cidr = net.IPNet{IP: net.ParseIP(toDelete), Mask: net.IPv4Mask(255, 255, 255, 255)}
+					ipv4Str = toDelete.String()
```

Comment:
Yeah, I can keep it as CidrInfo maybe(?). I did this in order to avoid conversion again to string array which is expected by `UnassignPrivateIpAddressesInput` . I can pass the CidrInfo and based on isPrefix we can derive the final caller.